File: Chapter 10/Connection.java
package chapter10;

public class Connection
{
    private Station station1, station2;
    private String lineName;
    
    public Connection(Station station1, Station station2, String lineName)
    {
        this.station1 = station1;
        this.station2 = station2;
        this.lineName = lineName;
    }

    public Station getStation1()
    {
        return station1;
    }

    public Station getStation2()
    {
        return station2;
    }

    public String getLineName()
    {
        return lineName;
    }
    
    public String toString()
    {
        return "[" + station1.getName() + ", " + station2.getName() + ", " + lineName + "]";
    }
    
}

File: Ch 10 - The OOA&D Lifecycle/Iteration 2 - Objectville Subway/SubwaySystem/Connection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Objectville_Subway.SubwaySystem
{
    class Connection
    {
        private Station _station1;
        private Station _station2;

        public string LineName { get; }
        public Station Station1 { get { return _station1; } }
        public Station Station2 { get { return _station2; } }

        public Connection(string lineName, Station station1, Station station2)
        {
            LineName = lineName;
            _station1 = station1;
            _station2 = station2;
        }
    }
}
